src/test/consensus/DistributedValidators_test.cpp

```diff
@@ -0,0 +1,321 @@
+    Copyright (c) 2012-2016 Ripple Labs Inc->
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+#include <BeastConfig.h>
+#include <ripple/beast/clock/manual_clock.h>
+#include <ripple/beast/unit_test.h>
+#include <test/csf.h>
+#include <utility>
+
+#include <boost/algorithm/string/split.hpp>
+#include <boost/algorithm/string/classification.hpp>
+#include <algorithm>
+#include <sstream>
+#include <fstream>
+#include <string>
+
+namespace ripple {
+namespace test {
+
+namespace csf
+{
+
+template<class SizePDF, class Generator>
+void
+randomRankedConnect(
+        PeerGroup &peers,
+        std::vector<double> const &ranks,
+        int numConnections,
+        SizePDF numConnectionsPDF,
+        std::chrono::milliseconds delay,
+        Generator &g)
+{
+    std::size_t const size = peers.size();
+    assert(size == ranks.size());
+
+    // 2. Generate UNLs based on sampling without replacement according
+    //    to weights.
+    std::vector<PeerGroup> connectionLists(numConnections);
+    std::vector<Peer *> rawPeers(peers.begin(), peers.end());
+    std::generate(connectionLists.begin(), connectionLists.end(), [&]() {
+        std::vector<Peer *> res =
+                random_weighted_shuffle(rawPeers, ranks, g);
+        res.resize(numConnectionsPDF(g));
+        return PeerGroup(std::move(res));
+    });
+
+    // 3. Assign trust
+    std::uniform_int_distribution<int> u(0, numConnections - 1);
+    for (auto &peer : peers)
+    {
+        for (auto &target : connectionLists[u(g)])
+            peer->connect(*target, delay);
+    }
+}
+
+} // csf
+
+class DistributedValidators_test : public beast::unit_test::suite
+{
+
+    void
+    completeTrustCompleteConnectFixedDelay(
+            std::size_t numPeers,
+            std::chrono::milliseconds delay = std::chrono::milliseconds(200),
+            bool printHeaders = false)
+    {
+        using namespace csf;
+        using namespace std::chrono;
+
+        // Initialize persistent collector logs specific to this method
+        static std::string const prefix =
+                "DistributedValidators::"
+                "completeTrustCompleteConnectFixedDelay";
+        static std::fstream
```

Comment:
In general I would agree, the behavior of static variables in functions or files is often something I have to look up, but in this case I don't see how it's an issue since the variable is const. I can easily remove the static and switch to append mode as you say though.